### DISCUSSION TITLE
Firefly III Data Importer 520 error

### DISCUSSION THREAD

**Discussion by git-deeper** [firefly-iii/firefly-iii#5568] [Firefly III]
Using https://data-importer.firefly-iii.org I am able to authenticate - Yay! - but then when choosing to import a CSV file, I don't get a chance to select the file before getting a 520 error:

Error 520 Ray ID: 6cf479dfbfd62cdd • 2022-01-18 02:43:30 UTC
Web server is returning an unknown error
You
Browser
Working
Chicago
Cloudflare
Working
data-importer.firefly-iii.org
Host
Error
What happened?
There is an unknown connection issue between Cloudflare and the origin web server. As a result, the web page can not be displayed.

Thanks for any suggestions!

**Comment by JC5** [comment]
I see no errors in my logs. Can you replace it with the same CSV file?

**Comment by cleon456** [comment]
I'm also getting the same error using the public data importer. Upon clicking import CSV file, it immediately switches to cloudflare's error 520 page. You don't even get to pick a file before it changes. And it stays in that page even after refreshing, going back, or opening a new tab with https://data-importer.firefly-iii.org. Only way to reaccess the data importer is to open up an incognito tab or clearing browser data.

**Comment by JC5** [comment]
I see some errors in the log.

```
[2022-01-21 07:34:04] local.ERROR: cURL error 35: error:1408F10B:SSL routines:ssl3_get_record:wrong version number (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://URL/oauth/token {"exception":"[object] (GuzzleHttp\\Exception\\ConnectException(code: 0): cURL error 35: error:1408F10B:SSL routines:ssl3_get_record:wrong version number (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://URL/oauth/token at /var/www/data-importer/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:210)
```

I've removed the url, but it contains the world 'cleon'.

**Reply by cleon456** [reply]
Ah yes, that was my dynamic dns address with the port that i used to test but that would just give an error 500. The error 520 would be using the actual url.

**Reply by JC5** [reply]
I've disabled the Cloudflare proxying for now, and will keep it off until after the weekend. Let me know if it helps. Maybe some time before it catches.

**Reply by JC5** [reply]
Apologies, later tonight this will happen.
